netty/src/main/java/io/grpc/netty/NettyClientTransport.java

```diff
@@ -194,6 +199,27 @@ public ClientStream newStream(
     if (channel == null) {
       return new FailingClientStream(statusExplainingWhyTheChannelIsNull, tracers);
     }
+    try {
+      if (callOptions.getAuthority() != null && !negotiator.mayBeVerifyAuthority(
+          callOptions.getAuthority())) {
+        String errMsg = String.format("Peer hostname verification failed for authority '%s'.",
+            callOptions.getAuthority());
+        logger.log(Level.FINE, errMsg);
+        return new FailingClientStream(Status.INTERNAL.withDescription(errMsg), tracers);
+      }
+    } catch (UnsupportedOperationException ex) {
+      logger.log(Level.FINE,
+          "Can't allow authority override in rpc when X509ExtendedTrustManager is not available.",
```

Comment:
This log doesn't seem to have a purpose, as we're about to fail the RPC itself. So the error is being handled.

Reply:
Done.